### DISCUSSION TITLE
Unreal Engine 5

### DISCUSSION THREAD

**Discussion by chrixbedardcad** [microdee/OpenUnrealSpaceMouse#51] [General]
Do you guys have any plan to make it working for UE5 or that already compatible?

**Comment by chrixbedardcad** [comment]
bty, I did copy the source under a /Plugins folder UE5 project, re-build and voila!, you guys are compatible with UE5 as far as I know :-)

**Reply by geeteecarnlo** [reply]
any chance of a link to the compiled plugin?, I'm on limited internet data so would be great not to have to get the UE5 source downloaded

**Comment by chrixbedardcad** [comment]
![image](https://user-images.githubusercontent.com/75350983/119873437-8bb21900-bef2-11eb-867b-872647c626d8.png)

**Comment by microdee** [comment]
I'm glad to hear, I didn't give SpaceMouse in UE5 a try yet. For precompiled release first I will need to make my build tool compatible with UE5 (atm my Unreal Locator doesn't find pre-installed binaries yet, but I hope it'll be an easy fix)

**Reply by geeteecarnlo** [reply]
thanks for the update microdee, you are fighting the good fight sir :) cheers

**Comment by OSC-Entity** [comment]
> bty, I did copy the source under a /Plugins folder UE5 project, re-build and voila!, you guys are compatible with UE5 as far as I know :-)

Hello! Any chance you could post the precompiled UE5 file? I can't seem to get to rebuild on my end :'(

Please please and thank you thank you!

**Reply by jameslashmar** [reply]
@chrixbedardcad +1 for the pre-built ue5 plugin please!

**Reply by OSC-Entity** [reply]
Since @microdee isn't responding to this I've found a tutorial for anyone trying to rebuild this plugin for UE5.  It fully worked for me! 

**Recompiling Plugins Tutorial**: https://youtu.be/WtfLfYY_k2Y

**Reply by geeteecarnlo** [reply]
thanks for the link, it got it to start (using the latest UE4-SpaceMouse-master) but after a few seconds it fails with this error:
ERROR: Could not find definition for module 'HIDUE', (referenced via command line -> SpaceMouse.uplugin)

I'm totally out of my depth when it comes to this kind of thing, is there anything obvious I am missing?

**Reply by geeteecarnlo** [reply]
I did try this, from a discord server:
**If you are trying to generate Visual Studio projects for your Unreal Engine 5 project, and running into an issue where UnrealBuildTool can't be found, open up a command prompt and use this to generate your projects instead (replace <PROJECT> with the FULL PATH to your .uproject file):
"C:\Program Files\Epic Games\UE_5.0EA\Engine\Binaries\DotNET\UnrealBuildTool\UnrealBuildTool.exe"  -projectfiles -project="<PROJECT>" -game -rocket -progress**
I applied it to the test project spacemouse.uproject and then was able to let it auto build after the usual warning, UE5 splash screen :) then warning message that it couldnt load spacemouse plugin :(

**Reply by geeteecarnlo** [reply]
Success, I was kindly shared this by a user on an Unreal Discord. at startup it was just moving in and out, opening up the 3dconnexion from the small icons to the right side of the taskbar(near the clock) I saw thats its default only had this one setting, turned them all on then restarted UE5 and boom, we're in, such a relief :)
https://wetransfer.com/downloads/b074154595ca73ec7120238f77a254e420210612104114/a9769d

**Reply by microdee** [reply]
@geeteecarnlo @OSC-Entity @jameslashmar please find the prebuilt binaries in the latest release https://github.com/microdee/UE4-SpaceMouse/releases/tag/1.2.4 [(direct download)](https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.4/SpaceMouse-Win64-1.2.4.5.0.0-PreBuilt.zip)

UE5 changed the path to couple of executables and I needed to update my build automation tool ( [Nuke.Unreal](https://github.com/microdee/Nuke.Unreal) ) in order to make it work.

Also in future situations like this I'd like to point out this section of the readme: https://github.com/microdee/UE4-SpaceMouse#build-a-release which points to https://github.com/microdee/UE4-SpaceMouse-TestProject where you can find exact instructions how to make a binary release for this plugin.

@geeteecarnlo This plugin doesn't use anything from 3DConnexion services. Configuring their own software won't affect anything in this plugin (yet). What you were experiencing is this mentioned in the README.md https://github.com/microdee/UE4-SpaceMouse#known-issues further explained in this issue: https://github.com/microdee/UE4-SpaceMouse/issues/12#issuecomment-696616414

**Reply by geeteecarnlo** [reply]
That great news microdee! many thanks!

**Comment by microdee** [comment]
I've uploaded prebuilt plugin for UE 5.0 you can find it in here: https://github.com/microdee/UE4-SpaceMouse/releases/tag/1.2.4 [(direct download)](https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.4/SpaceMouse-Win64-1.2.4.5.0.0-PreBuilt.zip)

In the same go I've uploaded the prebuilt plugin for 4.27 preview.

**Reply by MarkEHawkins** [reply]
I have downloaded the prebuilt plugin for UE 5.0. Do I install into my UE_5.0EA folder? The only folder I can see is .egstore. I have show hidden items checked. Thanks, I apricate your help.

**Reply by microdee** [reply]
this is what the zip should contain:

![image](https://user-images.githubusercontent.com/1009962/128902000-25ab5ec4-95ca-4ab1-88eb-e1fdbdcefc3a.png)

and this is where you put Unreal Engine plugins in their own folder (which will be used in all of your projects):

![ue-folders-sm](https://user-images.githubusercontent.com/1009962/128902172-9b95f8ad-452d-4277-8bed-c95da4aab9d6.jpg)

If your Unreal Engine installation doesn't have such folder structure then you should Verify the installation in your Epic Games launcher,

**Comment by MarkEHawkins** [comment]
Hi David, Thank very much for your help. All working now, folder mix-up on my part! Fantastic plugin. Thanks

**Comment by agencydhd** [comment]
Hi. This is great. Thank you for creating it. I had a conflict with the 3DConnexion drivers in the beginning but removed them and this works great. One quick question: Is there a way to make it less sensitive while getting used to it in Unreal? - if that is what the curve editors are for is there a little explanation of them? (if not i'll just play around with it and see.) thanks again!

**Reply by microdee** [reply]
You can set the maximum base translation and rotation speeds. The curves describe sensitivity based on how much you've pushed the puck from its resting position. So left side of the curve is the resting position and the right side is fully pushed. This way you can set up dead-zones or non-linear response curves. You can copy-paste them between translation and rotation by right clicking them.

**Reply by geeteecarnlo** [reply]
any chance someone has working binaries for Unreal 5 Preview?

**Reply by microdee** [reply]
Is this not working? https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.4/SpaceMouse-Win64-1.2.4.5.0.0-PreBuilt.zip (which is in https://github.com/microdee/UE4-SpaceMouse/releases/tag/1.2.4 since previous year august or something like that)

**Reply by geeteecarnlo** [reply]
Yeah, that was the first thing I tried (v1.2.4) but no luck

**Comment by aknittel1** [comment]
I went ahead and paid as well. 

Are you working on compatibility with UE 5 Preview, or at least providing them input if they broke the plug-in model? I would really this working this month.

**Reply by microdee** [reply]
Is this not working? https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.4/SpaceMouse-Win64-1.2.4.5.0.0-PreBuilt.zip (which is in https://github.com/microdee/UE4-SpaceMouse/releases/tag/1.2.4 since previous year august or something like that)

**Reply by hiabex** [reply]
Hi, I bought your plugin from the Epic store, ran into problems. Followed the breadcrumbs back to this Git thread, tried the above. The thing works like a beauty now. Thanks very much for making Spacemouse work for UE5. Money well spent.

(Sometime later) Not sure if this is plugin related, but I have found that since installing, Play Mode now doesn't work as usual. Directional keys are dead, prompt to shift+F1 to use mouse, but because the input of the engine is split between regular and Spacemouse, is it possible that level testing causes conflict? Pressing Esc, at least gets me back to build mode. I'd welcome your comments.

**Reply by microdee** [reply]
I have to test that, you could reproduce this in a simple project? If it's a consistent behavior and you think it's definitely caused by the SpaceMouse plugin you can make a Github issue for it to track it separately

**Reply by hiabex** [reply]
Hi, apologies for the late response. I haven't had a chance today to test an alternative project. I'll report back sometime tomorrow.

**Reply by hiabex** [reply]
Update: Ok, I'm going to guess that the reason that I got a freeze on Play mode was because I'd dropped a player actor into the level I'd made, wthout adding any blueprints. Rookie error. I tested the original plugin level and play mode works fine with the usual keys. Thanks again for you help and the beautifully integrated Spacemouse plugin.

**Comment by aknittel1** [comment]
Ill try that and report back to you.

**Comment by aknittel1** [comment]
This is what I get with UE 5 Preview 1:

![image](https://user-images.githubusercontent.com/22087392/156206361-898f83d3-1650-4fb8-87df-af6a1e1180f5.png)

**Reply by microdee** [reply]
Thanks for reporting, I'll investigate it soon and make a new prebuilt release for 5.0

**Reply by microdee** [reply]
try this? https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.4/SpaceMouse-Win64-1.2.4.5.0.0-PreBuilt.zip

**Reply by aknittel1** [reply]
Tried it. No issues works as expected.

**Reply by geeteecarnlo** [reply]
many thanks microdee, just tried it on my end and everthing is working well. cheers!

**Reply by KikiloCZ** [reply]
Hello, I tried the link above. I have Prewiev 2 installed and I received this error message. Thank you for your help.
![Výstřižek](https://user-images.githubusercontent.com/40862476/157395094-7eb61b7e-10c7-4b46-9b6e-05422c7cbce6.PNG)
.

**Comment by aknittel1** [comment]
Thanks!

**Comment by bond1mjb** [comment]
I just want to say thank you to the author of this plugin.   I have several 3DConnexion devices, and to my surprise, not only does this work on my newer Spacemouse Enterprise, but it also works flawlessly on my 17 year old original Spacepilot from 2005, wow!

As a newbie to Unreal Engine 5,  this has made learning the software so much more pleasant, thank you again for this wonderful plugin.
